test/optimizedForceMove.test.ts

```diff
@@ -0,0 +1,490 @@
+import {ethers} from 'ethers';
+import {expectRevert} from 'magmo-devtools';
+// @ts-ignore
+import optimizedForceMoveArtifact from '../build/contracts/TESTOptimizedForceMove.json';
+// @ts-ignore
+import countingAppArtifact from '../build/contracts/CountingApp.json';
+import {splitSignature, keccak256, defaultAbiCoder, arrayify} from 'ethers/utils';
+import {HashZero} from 'ethers/constants';
+
+let networkId;
+let optimizedForceMove: ethers.Contract;
+const provider = new ethers.providers.JsonRpcProvider(
+  `http://localhost:${process.env.DEV_GANACHE_PORT}`,
+);
+const signer = provider.getSigner(0);
+async function setupContracts() {
+  networkId = (await provider.getNetwork()).chainId;
+  const optimizedForceMoveContractAddress = optimizedForceMoveArtifact.networks[networkId].address;
+  optimizedForceMove = new ethers.Contract(
+    optimizedForceMoveContractAddress,
+    optimizedForceMoveArtifact.abi,
+    signer,
+  );
+}
```

Comment:
This file is very long. For the `ForceMove` contract it might be worth having a folder of tests, where each file tests a particular method.

Reply:
TODO added

Reply:
see https://github.com/statechannels/nitro-protocol/pull/15/commits/4b8cbb9de05dc2b91ee8abe45decde0508865bbd